README.md

```diff
@@ -37,22 +44,133 @@ api("com.github.spotify:openfeature-kotlin-sdk:[ANY_BRANCH]-SNAPSHOT")
 
 This will get a build from the head of the mentioned branch. 
 
-### Resolving a flag
+## 🌟 Features
+
+- support for various backend [providers](https://openfeature.dev/docs/reference/concepts/provider)
+- easy integration and extension via [hooks](https://openfeature.dev/docs/reference/concepts/hooks)
+- bool, string, numeric, and object flag types
+- [context-aware](https://openfeature.dev/docs/reference/concepts/evaluation-context) evaluation
+
+## 🚀 Usage
+
 ```kotlin
-import dev.openfeature.sdk.*
+    // configure a provider and get client
+    OpenFeatureAPI.setProvider(
+        CustomProvider.initialise()
+    )
+    val client = OpenFeatureAPI.getClient()
+
+    // get a bool flag value
+    client.getBooleanValue("boolFlag", default = false)
+    
+    // get a bool flag value async
+    coroutineScope.launch {
+        WithContext(Dispatchers.IO) {
+            awaitProviderReady()
+        }
+    // read properties
+    }
+```
+
+### Events
 
-// Change NoOpProvider with your actual provider
-OpenFeatureAPI.setProvider(NoOpProvider(), ImmutableContext())
-val flagValue = OpenFeatureAPI.getClient().getBooleanValue("boolFlag", false)
+Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
+Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are dispatched for every provider.
+Some providers support additional events, such as `PROVIDER_CONFIGURATION_CHANGED`.
+Please refer to the documentation of the provider you're using to see what events are supported.
+
+```kotlin
+    EventHandler.eventsObserver()
```

Comment:
I wonder if `eventsObserver` should be exposed from the `client` API or the `OpenFeatureAPI`, as it is [done in the Java SDK](https://github.com/open-feature/java-sdk#events)

Reply:
i just don’t know if putting it under OpenFeatureAPI makes it harder for provider to publish and listen for events. i will give it a try.